docker/qt6-build/Dockerfile

```diff
@@ -1,44 +1,111 @@
-FROM ubuntu:20.04 AS base
+ARG UBUNTU_VERSION=20.04
+ARG QT_MAJOR=6.5
+ARG QT_VERSION=6.5.6
+FROM ubuntu:${UBUNTU_VERSION} AS base
 LABEL maintainer="dan@dennedy.org"
 
 FROM base as build
+ARG UBUNTU_VERSION=20.04
+ARG QT_MAJOR=6.5
+ARG QT_VERSION=6.5.6
+
+# Debug: Show actual values being used (overrides from build-args)
+RUN echo "UBUNTU_VERSION override with build-arg to: ${UBUNTU_VERSION}" && \
+    echo "QT_MAJOR override with build-arg to: ${QT_MAJOR}" && \
+    echo "QT_VERSION override with build-arg to: ${QT_VERSION}"
 
 # Add deb-src lines to make apt-get build-dep work.
-RUN cat /etc/apt/sources.list | sed "s/deb /deb-src /" >> /etc/apt/sources.list
+# Handle both old format (20.04/22.04) and new deb822 format (24.04+)
+RUN if [ -f /etc/apt/sources.list.d/ubuntu.sources ]; then \
+      # New format: deb822 format in ubuntu.sources (24.04+)
+      sed 's/Types: deb$/Types: deb-src/' /etc/apt/sources.list.d/ubuntu.sources > /etc/apt/sources.list.d/ubuntu-src.sources; \
+    elif [ -f /etc/apt/sources.list ] && grep -q "^deb " /etc/apt/sources.list; then \
+      # Old format: traditional sources.list (20.04/22.04)
+      cat /etc/apt/sources.list | sed "s/deb /deb-src /" >> /etc/apt/sources.list; \
+    fi
 ARG DEBIAN_FRONTEND=noninteractive
-RUN apt-get -qq update && \
-  apt-get -yqq upgrade && \
-  apt-get -yqq build-dep qt5-default && \
-  apt-get -yqq install curl libxslt-dev libwebp-dev flex bison gperf ruby \
-    s3cmd libxcb1-dev libxcb-util-dev gpg software-properties-common \
-    g++ clang-10 libclang-10-dev libvulkan-dev ninja-build && \
-  curl https://apt.kitware.com/keys/kitware-archive-latest.asc | gpg --dearmor - >/etc/apt/trusted.gpg.d/kitware.gpg && \
-  apt-add-repository 'deb https://apt.kitware.com/ubuntu/ focal main' && \
-  apt-get update -qq && \
-  apt-get install -yqq cmake
+
+# Update system packages
+RUN apt-get -qq update && apt-get -yqq upgrade
+
+# Install Qt5 build dependencies - version specific
+RUN if [ "${UBUNTU_VERSION}" = "24.04" ]; then \
+      # Ubuntu 24.04: qt5-default doesn't exist, use qtbase5-dev instead
+      apt-get -yqq install qtbase5-dev qtbase5-dev-tools || apt-get -yqq install \
```

Comment:
Do not install qt5. `build-dep` is just a convenient way to get many dependencies since those are common between major versions. Maybe there is something like that better for qt6 on newer ubuntu.